adapters/impactify/impactify.go

```diff
@@ -0,0 +1,186 @@
+package impactify
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	"strings"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type ImpactifyExtBidder struct {
+	Impactify openrtb_ext.ExtImpImpactify `json:"impactify"`
+}
+
+type DefaultExtBidder struct {
+	Bidder openrtb_ext.ExtImpImpactify `json:"bidder"`
+}
+
+func (a *adapter) MakeRequests(bidRequest *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var validImps []openrtb2.Imp
+
+	var adapterRequests []*adapters.RequestData
+
+	for i := 0; i < len(bidRequest.Imp); i++ {
+		// Check if imp comes with bid floor amount defined in a foreign currency
+		if bidRequest.Imp[i].BidFloor > 0 && bidRequest.Imp[i].BidFloorCur != "" && strings.ToUpper(bidRequest.Imp[i].BidFloorCur) != "USD" {
+			// Convert to US dollars
+			convertedValue, err := reqInfo.ConvertCurrency(bidRequest.Imp[i].BidFloor, bidRequest.Imp[i].BidFloorCur, "USD")
+			if err != nil {
+				return nil, []error{err}
+			}
+			bidRequest.Imp[i].BidFloorCur = "USD"
+			bidRequest.Imp[i].BidFloor = convertedValue
+		}
+
+		var impactifyExt ImpactifyExtBidder
+
+		var defaultExt DefaultExtBidder
+		json.Unmarshal(bidRequest.Imp[i].Ext, &defaultExt)
+		impactifyExt.Impactify = defaultExt.Bidder
+		bidRequest.Imp[i].Ext, _ = json.Marshal(impactifyExt)
+
+		validImps = append(validImps, bidRequest.Imp[i])
+	}
+
+	if len(validImps) == 0 {
+		if errs != nil {
+			return nil, errs
+		} else {
+			return nil, []error{&errortypes.BadInput{
+				Message: "No valid impressions in the bid request",
+			}}
+		}
+	}
+
+	bidRequest.Imp = validImps
+
+	reqJSON, err := json.Marshal(bidRequest)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+	if bidRequest.Device != nil {
```

Comment:
Please add json tests for header values. There is currently no test coverage for any of the headers.

Reply:
Right, a new test covertage has been added.

Reply:
I do not see the test coverage added. There are no json tests which specify `device`, `site`, or `user` objects.

Reply:
Yes sorry, it is added now.

Reply:
No worries. Thank you.